crates/key-value/src/host_component.rs

```diff
@@ -66,4 +67,38 @@ impl DynamicHostComponent for KeyValueComponent {
 
         Ok(())
     }
+
+    fn validate_app(&self, app: &spin_app::App) -> anyhow::Result<()> {
+        let mut errors = vec![];
+
+        for component in app.components() {
+            let store_manager = self.manager.get(&component);
+            for allowed in component
+                .get_metadata(KEY_VALUE_STORES_KEY)?
+                .unwrap_or_default()
+            {
+                if !store_manager.is_defined(&allowed) {
+                    let err = format!("- Component {} uses store '{allowed}'", component.id());
+                    errors.push(err);
+                }
+            }
+        }
+
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            let prologue = vec![
+                "One or more components use key-value stores which are not defined.",
```

Comment:
I wonder if we need to explicitly call out that only the "default" store is implicitly defined. However, this is explained in the docs that are linked so i think this works and if there is confusion we can always iterate on user feedback

Reply:
Yeah, users should never see an error about the `default` store so hopefully they won't think they need to define it!.